pkgs/by-name/is/isisdl/package.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, fetchPypi
+, python3Packages
+, util-linux
+, withFFmpeg ? false, ffmpeg
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "isisdl";
+  version = "1.3.20";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-s0vGCJVSa6hf6/sIhzmaxpziP4izoRwcZfxvm//5inY=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
```

Comment:
```suggestion
  pyproject = true;
  
  dependencies = with python3Packages; [
```

Reply:
When applying your suggested change, `nix-build -A isisdl` leads to following errors:
```
Checking runtime dependencies for isisdl-1.3.20-py3-none-any.whl
  - cryptography~=38.0.4 not satisfied by version 42.0.5
  - requests~=2.28.1 not satisfied by version 2.31.0
  - packaging~=22.0 not satisfied by version 24.0
  - distro~=1.8.0 not satisfied by version 1.9.0
error: builder for '/nix/store/j0djw6zd8zzhg27n03qcxnkdmrmvhiz2-isisdl-1.3.20.drv' failed with exit code 1;
       last 10 log lines:
       > Successfully built isisdl-1.3.20-py3-none-any.whl
       > Finished creating a wheel...
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for isisdl-1.3.20-py3-none-any.whl
       >   - cryptography~=38.0.4 not satisfied by version 42.0.5
       >   - requests~=2.28.1 not satisfied by version 2.31.0
       >   - packaging~=22.0 not satisfied by version 24.0
       >   - distro~=1.8.0 not satisfied by version 1.9.0
```

This error only occurs when setting `pyproject = true`. Otherwise, everything builds fine. Unfortunately, I lack deeper knowledge of packaging. Could you help me out here? 

---

<details>
<summary><b>Git diff</b></summary>

```diff
diff --git a/pkgs/by-name/is/isisdl/package.nix b/pkgs/by-name/is/isisdl/package.nix
index 58589fa2d901..97d825533c3d 100644
--- a/pkgs/by-name/is/isisdl/package.nix
+++ b/pkgs/by-name/is/isisdl/package.nix
@@ -7,13 +7,15 @@
 python3Packages.buildPythonApplication rec {
   pname = "isisdl";
   version = "1.3.20";
+  pyproject = true;

   src = fetchPypi {
     inherit pname version;
     hash = "sha256-s0vGCJVSa6hf6/sIhzmaxpziP4izoRwcZfxvm//5inY=";
   };

-  propagatedBuildInputs = with python3Packages; [
+  dependencies = with python3Packages; [
+    setuptools
     cryptography
     requests
     pyyaml
```

</details>

Reply:
Ah, it looks like it should actually be

```diff
diff --git a/pkgs/by-name/is/isisdl/package.nix b/pkgs/by-name/is/isisdl/package.nix
index 58589fa2d901..3656c135eff1 100644
--- a/pkgs/by-name/is/isisdl/package.nix
+++ b/pkgs/by-name/is/isisdl/package.nix
@@ -6,31 +6,48 @@
 }:
 python3Packages.buildPythonApplication rec {
   pname = "isisdl";
   version = "1.3.20";
 
   src = fetchPypi {
     inherit pname version;
     hash = "sha256-s0vGCJVSa6hf6/sIhzmaxpziP4izoRwcZfxvm//5inY=";
   };
 
-  propagatedBuildInputs = with python3Packages; [
+  pyproject = true;
+
+  build-system = with python3Packages; [
+    setuptools
+  ];
+
+  nativeBuildInputs = with python3Packages; [
+    pythonRelaxDepsHook
+  ];
+
+  dependencies = with python3Packages; [
     cryptography
     requests
     pyyaml
     packaging
     colorama
     pyinotify
     distro
     psutil
   ];
 
+  pythonRelaxDeps = [
+    "cryptography"
+    "requests"
+    "packaging"
+    "distro"
+  ];
+
   buildInputs = [
```

Though I'm a little surprised that the dependency versions weren't checked without `pyproject = true`

Reply:
I can confirm. This works! Thank you!
I've applied and pushed the changes. :)